kalite/contentload/management/commands/channels/khan.py

```diff
@@ -79,39 +90,100 @@ def build_full_cache(items, id_key="id"):
     """
     for item in items:
```

Comment:
This function is too big. Sad to say that we're running out of time for 0.13, so let's just merge this. But I'd like this function to be a focus for breaking up into smaller chunks, and allowing for easier testing, especially since we've been getting a lot of bugs from this and `contentload` in general.